### PR TITLE
Fix when you remove the time from DatetimeWidget

### PR DESCRIPTION
**Pull request by iRohitSingh**
when you clear the time in datetimeWidget you will see a error.  This pr fixes it.

### CODE REVIEW

File: src/components/manage/Widgets/DatetimeWidget.jsx [hunk 1]
```diff
@@ -140,13 +140,13 @@ class DatetimeWidget extends Component {
       (prevState) => ({
         datetime: prevState.datetime
           ? prevState.datetime.set({
-              hours: time.hours(),
-              minutes: time.minutes(),
+              hours: time?.hours(),
+              minutes: time?.minutes(),
```
Comment (nzambello) [inline src/components/manage/Widgets/DatetimeWidget.jsx:144]: Wouldn't it be better with a default to 0?

```suggestion
              hours: time?.hours() ?? 0,
              minutes: time?.minutes() ?? 0,
```

File: src/components/manage/Widgets/DatetimeWidget.jsx [hunk 2]
```diff
@@ -140,13 +140,13 @@ class DatetimeWidget extends Component {
...
               seconds: 0,
             })
           : moment().set({
-              hours: time.hours(),
-              minutes: time.minutes(),
+              hours: time?.hours(),
+              minutes: time?.minutes(),
```
Comment (nzambello) [inline src/components/manage/Widgets/DatetimeWidget.jsx:149]: Wouldn't it be better with a default to 0?

```suggestion
              hours: time?.hours() ?? 0,
              minutes: time?.minutes() ?? 0,
```

**Comment by nzambello** [comment]
I don't remember how it was handled, maybe I suggested something stupid.

**Comment by tisto** [comment]
@iRohitSingh would it be possible to write a Cypress test that fails (in a separate PR maybe) and shows the issue?

**Comment by iRohitSingh** [comment]
@tisto  Ok, I will create

**Comment by iRohitSingh** [comment]
@tisto Cypress test that fails and shows the issue :https://github.com/plone/volto/pull/2950

File: src/components/manage/Widgets/DatetimeWidget.jsx (hunk 2 above)
Comment (iRohitSingh) [reply src/components/manage/Widgets/DatetimeWidget.jsx]: done

**Review by sneridagh — APPROVED** [review]

**Pull request merged**